src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -77,25 +111,30 @@ public StringProperty queryProperty() {
     }
 
     public void search() {
-        if (StringUtil.isBlank(getQuery())) {
+        String query = getQuery().trim();
+        if (StringUtil.isBlank(query)) {
             dialogService.notify(Localization.lang("Please enter a search string"));
             return;
         }
-
         if (stateManager.getActiveDatabase().isEmpty()) {
             dialogService.notify(Localization.lang("Please open or start a new library before searching"));
             return;
         }
 
         SearchBasedFetcher activeFetcher = getSelectedFetcher();
+        Globals.getTelemetryClient().ifPresent(client -> client.trackEvent("search", Map.of("fetcher", activeFetcher.getName()), Map.of()));
 
         BackgroundTask<ParserResult> task;
-        task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(getQuery().trim())))
-                             .withInitialMessage(Localization.lang("Processing %0", getQuery().trim()));
+        task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(query)))
```

Comment:
Don't you need to execute the task?
 .executeWith(Globals.TASK_EXECUTOR));

Reply:
I got you that you during review that there wasn't any code shown at your side.

Let me make a screenshot for you:

![grafik](https://user-images.githubusercontent.com/1366654/121432991-acab4d00-c97b-11eb-8122-1cad096b650c.png)